pkg/controllers/infracluster/vsphere.go

```diff
@@ -0,0 +1,204 @@
+/*
+Copyright 2024 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package infracluster
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+	"strconv"
+
+	"github.com/go-logr/logr"
+	mapiv1beta1 "github.com/openshift/api/machine/v1beta1"
+	"gopkg.in/yaml.v2"
+	corev1 "k8s.io/api/core/v1"
+	cerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	vspherev1 "sigs.k8s.io/cluster-api-provider-vsphere/apis/v1beta1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var (
+	errObtainingVSphereServerAddr             = errors.New("error obtaining VSphere server address")
+	errUnableToGetVSphereMAPIProviderSpec     = errors.New("unable to get VSphere MAPI ProviderSpec")
+	errUnableToFindCredsSecret                = errors.New("unable to get the VSphere credentials secret")
```

Comment:
I like the idea of using known errors for these credential cases, since callers could switch on the error returned. However, for that to be useful in code, these would need to be exported.

Reply:
I created these and the ones in infracluster_controller.go because they are required by the new linter we added. 
I don't necessarily thing this package will be used anywhere else other than through the manager, hence I don't think it is worth or useful exporting the errs.